### PR TITLE
ref(proj-config): Wrap scheduling in a sentry span

### PR DESCRIPTION
**Pull request by iker-barriocanal**
Updating a lot of organizations or projects in a single database transaction causes the `on_commit` list to grow considerably and may cause memory leaks. It may be very hard to identify these cases.

In an attempt to make these simpler, the `on_commit` call is surrounded by a sentry span to make it easier to identify on Sentry.

### CODE REVIEW

File: src/sentry/tasks/relay.py [hunk 1]
```diff
@@ -270,15 +270,21 @@ def schedule_invalidate_project_config(
...
-    transaction.on_commit(
-        lambda: _schedule_invalidate_project_config(
-            trigger=trigger,
-            organization_id=organization_id,
-            project_id=project_id,
-            public_key=public_key,
-            countdown=countdown,
+    with sentry_sdk.start_span(
+        op="relay.projectconfig_cache.invalidation.schedule_after_db_transaction",
+    ):
```
Comment (jjbayer) [inline src/sentry/tasks/relay.py:275]: Is it possible to search for spans in sentry? If so, how would you find outliers that have a lot of these spans for a single transaction?

**Review by untitaker — COMMENTED** [review]
The idea was to wrap the callback that is passed to on_commit, because that one can be expensive, not the code that registers the callback

To joris question, this would be to fill missing instrumentation in transactions across the codebase, so more for product teams than ingest

File: src/sentry/tasks/relay.py (hunk 1 above)
Comment (iker-barriocanal) [reply src/sentry/tasks/relay.py:275]: It's currently not possible. Given the likeliness of this issue happening, and the fact that transactions in sentry represent API endpoints and tasks, the combination of the increased visibility this span provides plus statsd metrics should make identifying the problem easy enough not to use a transaction.

**Review by untitaker — APPROVED** [review]
ah right, we want to add spans in case ppl run out of memory. all good

**Pull request merged**